driver/src/main.rs

```diff
@@ -202,10 +205,15 @@ fn main() {
     let price_finder = price_finding::create_price_finder(fee, options.solver_type, price_oracle);
 
     // Create the orderbook reader.
-    let orderbook = PaginatedStableXOrderBookReader::new(&contract, options.auction_data_page_size);
-    info!("Orderbook filter: {:?}", options.orderbook_filter);
+    let primary_orderbook =
+        PaginatedStableXOrderBookReader::new(contract.clone(), options.auction_data_page_size);
```

Comment:
Instead of cloning the contract would it be better to use an RC here instead?

Reply:
I think we would have to use `Arc`, but that would work.

Reply:
Actually, this is something that `graphprotocol` did a lot, they used `Arc` quite extensively and `Mutex`es for interior mutability. I think it definitely makes some code easier to write (no need to reason about lifetimes for example) and has a fairly low runtime cost.

Maybe we can use to use `Arc` in a some places where it makes sense and remove some of the `'a` references we carry around.

Reply:
I'm a major fan of these types of simplifications. If only, I knew why/when clone vs RC vs Arc and why they make it easier.